src/gen.rs

```diff
@@ -82,6 +84,17 @@ impl Generator {
     /// * Appends generated Rust types to the output
     fn gen_in_order(&self, deps: &mut Vec<Schema>, output: &mut impl Write) -> Result<()> {
         let mut gs = GenState::new();
+        let mut schemata_by_name: HashMap<Name, Schema> = HashMap::new();
```

Comment:
I think `schemata_by_name` could be part of `GenState`.

Then current `GenState::new()` would become `GenState::new(deps)` with `pub fn new(deps: &[Schema])`.
And its current `HashMap<String, String>` would become a specific field `types_by_schema`.

Reply:
Done! 
Please review!

What would be a better name for the `inner` field (the old HashMap) ?

Reply:
I think `types_by_schema` would be good.

Reply:
I didn't read carefully your first comment!
I will rename the fields now!

Reply:
Fixed!